File: backend/tests/factories/pricing.py
import factory
from faker import Faker

from apps.pricing.models import Price, PriceRank, PriceRankTranslation, PriceTranslation
from tests.factories.language import LanguageFactory

faker = Faker()


class PriceFactory(factory.django.DjangoModelFactory):
    """Factory for Price model."""

    class Meta:
        model = Price

    type = factory.Sequence(lambda n: f"price_type_{n}")
    visibility = factory.Iterator(["public", "hidden", "members"])
    membership = factory.Iterator(["", "member", "vip"])
    minimum = None
    maximum = None
    step = None
    sort_order = factory.Sequence(lambda n: n)
    cineville_box = False


class PriceTranslationFactory(factory.django.DjangoModelFactory):
    """Factory for PriceTranslation model."""

    class Meta:
        model = PriceTranslation

    price = factory.SubFactory(PriceFactory)
    language = factory.SubFactory(LanguageFactory)
    description = factory.LazyAttribute(lambda _: faker.sentence(nb_words=6))


class PriceRankFactory(factory.django.DjangoModelFactory):
    """Factory for PriceRank model."""

    class Meta:
        model = PriceRank

    position = factory.Sequence(lambda n: n + 1)
    sold_out_buffer = 0


class PriceRankTranslationFactory(factory.django.DjangoModelFactory):
    """Factory for PriceRankTranslation model."""

    class Meta:
        model = PriceRankTranslation

    price_rank = factory.SubFactory(PriceRankFactory)
    language = factory.SubFactory(LanguageFactory)
    description = factory.LazyAttribute(lambda _: faker.sentence(nb_words=6))

File: frontend/src/types/Pricing.ts
/**
 * Price object returned by the backend `/prices/` endpoints.
 */
export interface Price {
  id: number
  type: string
  visibility: string
  membership: string
  minimum: number | null
  maximum: number | null
  step: number | null
  sort_order: number
  cineville_box: boolean
  description: Record<string, string> | null
  display_description: string | null
}

/**
 * PriceRank object returned by the backend `/price-ranks/` endpoints.
 */
export interface PriceRank {
  id: number
  position: number
  sold_out_buffer: number
  description: Record<string, string> | null
  display_description: string | null
}

/**
 * Paginated response shape for `GET /prices/`.
 */
export interface PriceListResponse {
  results: Price[]
  count?: number
  next?: string | null
  previous?: string | null
}

/**
 * Paginated response shape for `GET /price-ranks/`.
 */
export interface PriceRankListResponse {
  results: PriceRank[]
  count?: number
  next?: string | null
  previous?: string | null
}
